packages/lesswrong/components/ultraFeed/UltraFeedThreadItem.tsx

```diff
@@ -0,0 +1,190 @@
+import { useTracking } from "../../lib/analyticsEvents";
+import { defineStyles, useStyles } from "../hooks/useStyles";
+import { DisplayFeedCommentThread } from "./ultraFeedTypes";
+
+const styles = defineStyles("UltraFeedThreadItem", (theme: ThemeType) => ({
+  root: {
+    paddingLeft: 16,
+    paddingRight: 16,
+    borderRadius: 4,
+    backgroundColor: theme.palette.panelBackground.default,
+  },
+  commentsContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    gap: 16,
+  },
+  commentsList: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  commentItem: {
+    position: 'relative',
+  },
+}));
+
+
+/* if there are multiple comments in a row that are collapsed, compress them into a single placeholder */
+function compressCollapsedComments(
+  displayStatuses: Record<string, "expanded" | "collapsed" | "hidden">,
+  comments: UltraFeedComment[],
+) {
+  const result: Array<UltraFeedComment | { placeholder: true; hiddenComments: UltraFeedComment[] }> = [];
+
+  if (comments.length === 0) return result;
+
+  if (comments.length > 0) {
+    result.push(comments[0]);
+  }
+
+  let tempGroup: UltraFeedComment[] = [];
+
+  const flushGroupIfNeeded = () => {
+    if (tempGroup.length >= 2) {
+      result.push({ placeholder: true, hiddenComments: [...tempGroup] });
+    } else {
+      tempGroup.forEach(item => result.push(item));
+    }
+    tempGroup = [];
+  };
+
+  for (let i = 1; i < comments.length; i++) {
+    const comment = comments[i];
+    const commentId = comment._id;
+    const localStatus = displayStatuses[commentId] || "collapsed"
+
+    if (localStatus === "hidden") {
+      continue;
+    }
+
+    if (localStatus === "collapsed") {
+      tempGroup.push(comment);
+    } else {
+      // If we hit a non-collapsed, flush the current group
+      flushGroupIfNeeded();
+      result.push(comment);
+    }
+  }
+  flushGroupIfNeeded();
+
+  return result;
+}
+
+const UltraFeedThreadItem = ({thread}: {
+  thread: DisplayFeedCommentThread,
+}) => {
+  const { comments, commentMetaInfos } = thread;
+
+  const classes = useStyles(styles);
+  const {captureEvent} = useTracking();
+
+  const [commentDisplayStatuses, setCommentDisplayStatuses] = useState<Record<string, "expanded" | "collapsed" | "hidden">>(() => {
+    // Initialize from commentMetaInfos if available
+    const result: Record<string, "expanded" | "collapsed" | "hidden"> = {};
+    
+    for (const [commentId, meta] of Object.entries(commentMetaInfos || {})) {
+      // For the first comment, ensure it's at least "collapsed"
+      if (comments.length > 0 && commentId === comments[0]._id) {
+        const firstCommentStatus = meta.displayStatus === "hidden"
+        ? "collapsed"
+        : meta.displayStatus ?? "collapsed";
+        result[commentId] = firstCommentStatus;
+      } else {
+        result[commentId] = meta.displayStatus || "collapsed";
+      }
+    }
+
+    return result;
+  });
```

Comment:
I don't love large inline `useState` initializers, would prefer it to be extracted to a named function indicating what it's doing